common/persistence/persistenceRateLimitedClients.go

```diff
@@ -1029,3 +1046,14 @@ func allow(
 		callerInfo.CallOrigin,
 	))
 }
+
+// TODO: change the value return so that it can also be used by
+// persistence metrics client. For now, it's only used by rate
+// limit client, and we don't really care about the actual value
+// returned, as long as they are different from each task category.
+func ConstructHistoryTaskAPI(
```

Comment:
persistence retry/metrics/ratelimit client should be moved to `persistence/client` package. Then this function doesn't have to be exported.